File: samples/PHP/runtime-evaluation-in-an-environment.php
<?php
function eval_with_x($code, $a, $b) {
    $x = $a;
    $first = eval($code);
    $x = $b;
    $second = eval($code);
    return $second - $first;
}

echo eval_with_x('return 3 * $x;', 5, 10), "\n"; # Prints "15".
?>

File: samples/Ruby/runtime-evaluation-in-an-environment.rb
def bind_x_to_value(x)
  binding
end

def eval_with_x(code, a, b)
  eval(code, bind_x_to_value(b)) - eval(code, bind_x_to_value(a))
end

puts eval_with_x('2 ** x', 3, 5) # Prints "24"
